pkg/build/controller/controller.go

```diff
@@ -289,6 +291,81 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			return fmt.Errorf("failed to update build %s/%s: %v", build.Namespace, build.Name, err)
 		}
 		glog.V(4).Infof("Build %s/%s status was updated %s -> %s", build.Namespace, build.Name, build.Status.Phase, nextStatus)
+
+		if buildutil.IsBuildComplete(build) {
+			if err := bc.executeBuildPostHooks(build); err != nil {
+				// Do not return the error, since errors executing the
+				// post hooks should not influence the build lifecycle.
+				util.HandleError(fmt.Errorf("failed to execute post hooks of build %s/%s: %v", build.Namespace, build.Name, err))
+			} else {
+				glog.V(4).Infof("Post hooks of build %s/%s were executed successfully", build.Namespace, build.Name)
+			}
+		}
+	}
```

Comment:
I think we need controller logic to also re-process builds to ensure the post-actions get invoked, but maybe not, if this is the only code that can update a build to complete/failed, and it's also responsible for invoking the actions, that could be sufficient.